src/ProviderPR/SFA.DAS.ProviderPR/Application/Queries/GetEASUserByEmail/GetEasUserByEmailQueryHandler.cs

```diff
@@ -0,0 +1,45 @@
+﻿using MediatR;
+using SFA.DAS.SharedOuterApi.Configuration;
+using SFA.DAS.SharedOuterApi.InnerApi.Requests;
+using SFA.DAS.SharedOuterApi.InnerApi.Requests.EmployerAccounts;
+using SFA.DAS.SharedOuterApi.InnerApi.Responses;
+using SFA.DAS.SharedOuterApi.InnerApi.Responses.EmployerAccounts;
+using SFA.DAS.SharedOuterApi.Interfaces;
+using System.Net;
+
+namespace SFA.DAS.ProviderPR.Application.Queries.GetEasUserByEmail;
+public class GetEasUserByEmailQueryHandler(IAccountsApiClient<AccountsConfiguration> apiClient) : IRequestHandler<GetEasUserByEmailQuery, GetEasUserByEmailQueryResult>
+{
+    public async Task<GetEasUserByEmailQueryResult> Handle(GetEasUserByEmailQuery request, CancellationToken cancellationToken)
+    {
+        var result = await apiClient.GetWithResponseCode<GetUserByEmailResponse>(new GetUserByEmailRequest(request.Email));
+
+        if (result.StatusCode != HttpStatusCode.OK)
+        {
+            return new GetEasUserByEmailQueryResult(false, null, null, null);
+        }
+
+        var userRef = result.Body.Ref;
+
+        var accounts = await apiClient.GetAll<GetUserAccountsResponse>(new GetUserAccountsRequest(userRef.ToString()));
+
+        if (!accounts.Any())
+        {
+            return new GetEasUserByEmailQueryResult(false, null, null, null);
+        }
+
+        if (accounts.Count() > 1)
+        {
+            return new GetEasUserByEmailQueryResult(true, false, null, null);
+        }
+
+        var account = accounts.First();
+        var accountId = account!.AccountId;
```

Comment:
Should not need a null negator here as First() will always return not null value or throw exception.

Reply:
fixed